spackbot/handlers/pipelines.py

```diff
@@ -52,20 +52,43 @@ async def run_pipeline(event, gh):
     branch = pr["head"]["ref"]
     branch = f"github/pr{number}_{branch}"
     branch = urllib.parse.quote_plus(branch)
-
-    url = f"{helpers.gitlab_spack_project_url}/pipeline?ref={branch}"
-    headers = {"PRIVATE-TOKEN": GITLAB_TOKEN}
-
-    logger.info(f"{sender} triggering pipeline, url = {url}")
-
-    # Don't provide GitHub credentials to GitLab!
-    async with aiohttp.ClientSession() as session:
-        async with session.post(url, headers=headers) as response:
-            result = await response.json()
-
+    if not retry_pipeline:
+        trigger_type = "started"
+        url = f"{helpers.gitlab_spack_project_url}/pipeline?ref={branch}"
+        headers = {"PRIVATE-TOKEN": GITLAB_TOKEN}
+
+        # Don't provide GitHub credentials to GitLab!
+        async with aiohttp.ClientSession() as session:
+            async with session.post(url, headers=headers) as response:
+                result = await response.json()
+    else:
+        # Get a list of failed pipelines
+        trigger_type = "retried"
+        query_string = f"?status=failed&order_by=updated_at&ref={branch}"
+        url = f"{helpers.gitlab_spack_project_url}/pipelines{query_string}"
+        headers = {"PRIVATE-TOKEN": GITLAB_TOKEN}
+
+        # Don't provide GitHub credentials to GitLab!
+        async with aiohttp.ClientSession() as session:
+            async with session.post(url, headers=headers) as response:
+                pipelines = await response.json()
+        logger.info(f"{sender} triggering pipeline, url = {url}")
```

Comment:
I'm aware that you didn't add this line, but just moved it around from where it was before.  But in the spackbot I'm currently looking to get merged (#58), I just discovered that `sender` is a ginormous json object, so it might make sense to use `debug` here instead of `info`.  Part of my PR allows to set the desired log level using an environment variable, so it should be easy enough to turn on that firehose if we need.

Reply:
That's an easy fix.  I'll do so.